### PR TITLE
feat: Diagnostics CLI 2.5.0 new features and release notes

### PR DESCRIPTION
**Pull request by daffinito**
Please follow [conventional commit standards](https://www.conventionalcommits.org/en/v1.0.0/)
in your commit messages and pull request title.

## Give us some context

* What problems does this PR solve?

We are releasing a new version of the Diagnostics CLI which includes a new command line flag `-api-key`.  This PR includes the updates to the relevant docs to document the new flag, and the release notes for the new cli version.

* Add any context that will help us review your changes such as testing notes,
  links to related docs, screenshots, etc.
* If your issue relates to an existing GitHub issue, please link to it.

### CODE REVIEW

**Comment by daffinito** [comment]
hi @ally-sassman , this is ready for review now. thank you!

File: src/content/docs/new-relic-solutions/solve-common-issues/diagnostics-cli-nrdiag/interpret-nrdiag-output.mdx [hunk 1]
```diff
@@ -155,7 +155,7 @@ The Diagnostics CLI returns the following status codes after running:
 
 You can examine the `nrdiag-output.json` output in the UI. The account you select is associated with the license key that was validated by the Diagnostics CLI for upload.
 
-1. Select an app in [in the UI](https://one.newrelic.com), then search for **Diagnostics CLI Output**. OR, click this [**Diagnostics CLI Output** permalink](https://onenr.io/0x0jldL3EjW) to go directly to the page.
+1. Select an app [in the UI](https://one.newrelic.com), then search for **Diagnostics CLI Output**. OR, click this [**Diagnostics CLI Output** permalink](https://onenr.io/0x0jldL3EjW) to go directly to the page.
```
Comment (ally-sassman) [inline src/content/docs/new-relic-solutions/solve-common-issues/diagnostics-cli-nrdiag/interpret-nrdiag-output.mdx:158]:
```suggestion
1. Select an app [in the UI](https://one.newrelic.com), then search for **Diagnostics CLI Output**.
```
I feel like we should offer only one method to get to a page, even if there are other ways to get there. This method allows them to first pick their relevant app, vs. the option which is unclear which app it opens with.

File: src/content/docs/new-relic-solutions/solve-common-issues/diagnostics-cli-nrdiag/pass-command-line-options-nrdiag.mdx [hunk 2]
```diff
@@ -97,7 +107,7 @@ To use the following command line options with the Diagnostics CLI:
       </td>
 
       <td>
-        Include a file or directory (including subdirectories) in the `nrdiag-output.zip`. Limit 4GB. To upload the results to New Relic also use the `-a` flag.
+        Include a file or directory (including subdirectories) in the `nrdiag-output.zip`. Limit 4GB. To upload the results to New Relic also use the `-a` or `-api-key` flag.
```
Comment (ally-sassman) [inline src/content/docs/new-relic-solutions/solve-common-issues/diagnostics-cli-nrdiag/pass-command-line-options-nrdiag.mdx:110]:
```suggestion
        Include a file or directory (including subdirectories) in the `nrdiag-output.zip`. Limit 4GB. Use the `-a` or `-api-key` flag to upload the results to New Relic.
```

File: src/content/docs/new-relic-solutions/solve-common-issues/diagnostics-cli-nrdiag/run-diagnostics-cli-nrdiag.mdx [hunk 3]
```diff
@@ -244,44 +244,82 @@ To run from PowerShell, add `./` to the start of `cmd`.
 ## Upload results to a New Relic account [#attach-account-results]
 
 <Callout variant="important">
-  If your system is configured to **not** connect to external IP addresses, this method will not work. Instead, attach the [output files](/docs/using-new-relic/cross-product-functions/diagnostics-cli-nrdiag/interpret-nrdiag-output) in an email to New Relic Support. If there is an issue with your <InlinePopover type="licenseKey" />, you will also need to send your output files in an email to support.
+  If your system is configured to **not** connect to external IP addresses, this method will not work. Instead, attach the [output files](/docs/using-new-relic/cross-product-functions/diagnostics-cli-nrdiag/interpret-nrdiag-output) in an email to New Relic Support.
```
Comment (ally-sassman) [inline src/content/docs/new-relic-solutions/solve-common-issues/diagnostics-cli-nrdiag/run-diagnostics-cli-nrdiag.mdx:247]:
```suggestion
  If your system is **not** configured to connect to external IP addresses, this method will not work. Instead, attach the [output files](/docs/using-new-relic/cross-product-functions/diagnostics-cli-nrdiag/interpret-nrdiag-output) in an email to New Relic Support.
```

File: src/content/docs/new-relic-solutions/solve-common-issues/diagnostics-cli-nrdiag/run-diagnostics-cli-nrdiag.mdx [hunk 4]
```diff
@@ -244,44 +244,82 @@ To run from PowerShell, add `./` to the start of `cmd`.
...
 </Callout>
 
-### Automatic Account upload [#automatic]
+### Automatic account upload [#automatic]
 
-To upload your results automatically to a New Relic account when the Diagnostics CLI is executed, use the `-attach` or `-a` command line flag. This will validate any New Relic License Keys found in your environment for upload.
+To upload your results automatically to a New Relic account when the Diagnostics CLI is executed, use either the `-attach` or `-api-key` command line flags. The `-attach` flag will validate any New Relic License Keys found in your environment, and upload the run to the account associated with the license key. Alternatively, the `-api-key` flag will use the <InlinePopover type="userKey" /> provided to upload the run, regardless of the presense of a valid license key.
```
Comment (ally-sassman) [inline src/content/docs/new-relic-solutions/solve-common-issues/diagnostics-cli-nrdiag/run-diagnostics-cli-nrdiag.mdx:252]:
```suggestion
To upload your results automatically to a New Relic account when the Diagnostics CLI is executed, use either the `-attach` or `-api-key` command line flags. The `-attach` flag will validate any New Relic license keys found in your environment and upload the run to the account associated with the license key. Alternatively, the `-api-key` flag will use the <InlinePopover type="userKey" /> provided to upload the run, regardless of the presence of a valid license key.
```

File: src/content/docs/release-notes/diagnostics-release-notes/diagnostics-cli-release-notes/diagnostics-cli-250.mdx [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+version: 2.5.0
+downloadLink: 'https://download.newrelic.com/nrdiag/nrdiag_2.5.0.zip'
+---
+
+## Changes
+### New Feature
+- `-api-key` command line flag
+  - Use your <InlinePopover type="userKey" /> to automatically attach a Diagnostics CLI run to your account. This bypasses the requirement for a validated license key to attach a run.
+  - Example usage: `./nrdiag -api-key NRAK-123...............`
+  - [Read more here](https://docs.newrelic.com/docs/new-relic-solutions/solve-common-issues/diagnostics-cli-nrdiag/run-diagnostics-cli-nrdiag/#automatic-api)
```
Comment (ally-sassman) [inline src/content/docs/release-notes/diagnostics-release-notes/diagnostics-cli-release-notes/diagnostics-cli-250.mdx:13]:
```suggestion
  - [Read more here](/docs/new-relic-solutions/solve-common-issues/diagnostics-cli-nrdiag/run-diagnostics-cli-nrdiag/#automatic-api)
```

File: src/content/docs/release-notes/diagnostics-release-notes/diagnostics-cli-release-notes/diagnostics-cli-250.mdx [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+### Python Agent
```
Comment (ally-sassman) [inline src/content/docs/release-notes/diagnostics-release-notes/diagnostics-cli-release-notes/diagnostics-cli-250.mdx:15]:
```suggestion
### Python agent
```

File: src/content/docs/release-notes/diagnostics-release-notes/diagnostics-cli-release-notes/diagnostics-cli-250.mdx [hunk 7]
```diff
@@ -0,0 +1,18 @@
+---
+subject: Diagnostics CLI (nrdiag)
+releaseDate: '2023-05-04'
+version: 2.5.0
+downloadLink: 'https://download.newrelic.com/nrdiag/nrdiag_2.5.0.zip'
+---
+
+## Changes
+### New Feature
```
Comment (ally-sassman) [inline src/content/docs/release-notes/diagnostics-release-notes/diagnostics-cli-release-notes/diagnostics-cli-250.mdx:9]:
```suggestion
### New features
```

File: src/content/docs/release-notes/diagnostics-release-notes/diagnostics-cli-release-notes/diagnostics-cli-250.mdx (hunk 5 above)
Comment (ally-sassman) [inline src/content/docs/release-notes/diagnostics-release-notes/diagnostics-cli-release-notes/diagnostics-cli-250.mdx:13]:
```suggestion
`-api-key` command line flag
  * Use your <InlinePopover type="userKey" /> to automatically attach a Diagnostics CLI run to your account. This bypasses the requirement for a validated license key to attach a run.
  * Example usage: `./nrdiag -api-key NRAK-123...............`
  * [Read more here](/docs/new-relic-solutions/solve-common-issues/diagnostics-cli-nrdiag/run-diagnostics-cli-nrdiag/#automatic-api)
```

File: src/content/docs/release-notes/diagnostics-release-notes/diagnostics-cli-release-notes/diagnostics-cli-250.mdx (hunk 7 above)
Comment (ally-sassman) [inline src/content/docs/release-notes/diagnostics-release-notes/diagnostics-cli-release-notes/diagnostics-cli-250.mdx:9]:
```suggestion
### New feature
```

File: src/content/docs/release-notes/diagnostics-release-notes/diagnostics-cli-release-notes/diagnostics-cli-250.mdx [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+- `-api-key` command line flag
+  - Use your <InlinePopover type="userKey" /> to automatically attach a Diagnostics CLI run to your account. This bypasses the requirement for a validated license key to attach a run.
+  - Example usage: `./nrdiag -api-key NRAK-123...............`
+  - [Read more here](https://docs.newrelic.com/docs/new-relic-solutions/solve-common-issues/diagnostics-cli-nrdiag/run-diagnostics-cli-nrdiag/#automatic-api)
+
+### Python Agent
+- **Python/Env/Version**: Add check for output of `python3 –version`
+- **Python/Env/Dependencies**: Add check for output of `pip3 –version`
+- **Python/Requirements/PythonVersion:** Update New Relic Agent Supportability for Python 3.6, 3.9, 3.10, and 3.11.
```
Comment (ally-sassman) [inline src/content/docs/release-notes/diagnostics-release-notes/diagnostics-cli-release-notes/diagnostics-cli-250.mdx:18]:
```suggestion
- **Python/Requirements/PythonVersion:** Update New Relic agent supportability for Python 3.6, 3.9, 3.10, and 3.11.
```

File: src/content/docs/new-relic-solutions/solve-common-issues/diagnostics-cli-nrdiag/interpret-nrdiag-output.mdx [hunk 9]
```diff
@@ -155,7 +155,7 @@ The Diagnostics CLI returns the following status codes after running:
...
 2. Optional: Click the star next to the icon to add it to your favorites.
```
Comment (ally-sassman) [inline src/content/docs/new-relic-solutions/solve-common-issues/diagnostics-cli-nrdiag/interpret-nrdiag-output.mdx:159]:
```suggestion
2. (Optional) Click the star next to the icon to add it to your favorites.
```

**Review by ally-sassman — CHANGES_REQUESTED** [review]
Reviewed with a few suggested changes.

**Comment by daffinito** [comment]
@ally-sassman Thank you for the suggestions! I committed them!

File: src/content/docs/release-notes/diagnostics-release-notes/diagnostics-cli-release-notes/diagnostics-cli-250.mdx [hunk 10]
```diff
@@ -0,0 +1,18 @@
+---
+subject: Diagnostics CLI (nrdiag)
+releaseDate: '2023-05-04'
```
Comment (ally-sassman) [inline src/content/docs/release-notes/diagnostics-release-notes/diagnostics-cli-release-notes/diagnostics-cli-250.mdx:3]:
```suggestion
releaseDate: '2023-05-08'
```
Should this be changed to today's date?

Comment (daffinito) [reply src/content/docs/release-notes/diagnostics-release-notes/diagnostics-cli-release-notes/diagnostics-cli-250.mdx:3]: oh yeah probably, thank you!

**Review by ally-sassman — APPROVED** [review]

**Pull request merged**